app/src/main/java/com/example/mystore/viewmodels/ProductViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.mystore.viewmodels
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.example.mystore.ProductsRepository
+import com.example.mystore.models.ProductModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class ProductViewModel @Inject constructor(
+    private val repository: ProductsRepository
+) : ViewModel() {
+    var name: String by mutableStateOf("")
+    var description: String by mutableStateOf("")
+    var category: String by mutableStateOf("")
+    var price: Float by mutableStateOf(0.0f)
+    var rating: Int by mutableStateOf(0)
+    fun getProduct(): ProductModel {
+//        var product = repository.getProduct()
+        name = "Stargizer's Tea Set"
+        description =
+            "Include in our heavenly tea experience with our Stargizer's Tea Set, featuring a constellation-themed teapot and two matching teacups. Crafted from fine porcelain, this elegant set will transport you to the cosmos with every sip."
+        category = "Home"
+        price = 90.00f
+        rating = 5
+        return ProductModel(name, description, category, price, rating)
```

Comment:
You can just store a `ProductModel` object as state in the view model, instead of all of its components separately.